### ISSUE TITLE
Export (geometry & appearance) Not Like Blender Original

### ISSUE THREAD

**Issue by DrX3D** [KhronosGroup/glTF-Blender-Exporter#51]
I took the HOUSE model from Blend Source (https://www.blendswap.com/blends/view/15091) to try out the exporter. I opened it in the latest release of Blender (V2.78c) with the export code downloaded and installed yesterday (2017-07-20). I exported the model to a binary glTF format and loaded it in Don McCurdy's glTF viewer (https://gltf-viewer.donmccurdy.com/). 

Original:
![blender](https://user-images.githubusercontent.com/8261607/28474618-33b522ae-6dfe-11e7-9f8e-7c05a0012279.jpg)

Binary glTF
![gltfbinary](https://user-images.githubusercontent.com/8261607/28474826-e074708a-6dfe-11e7-9722-f7aec0fe2866.jpg)

The results are not the same.
Geometry
* The front steps and roof are not on flat piece. It appears that the normals may be reversed.
* There is something unexpected going on at ground level under the 4 front windows.
* Roof geometry appears inverted. This may be the same problem as the first bullet

Appearance:
* The Blender model is flat colored (not necessarily a flat shading), while the one in the glTF viewer is very reflective.
* The windows are transparent in Blender and interior detail is visible, not in glTF

The text-based glTF file does not display at all in the viewer. I also exported in X3D and used BitManagement's BS Contact viewer. That model & display shows the same geometry issues, but is closer to Blender in appearance.
![x3d](https://user-images.githubusercontent.com/8261607/28474939-4e23766c-6dff-11e7-8aaf-e2341f0c046a.jpg)

I am not sure if the problem is the exporter or the glTF2 loader (or even something else), but this seemed the best place to start. I am not really Blender-knowledgeable so the problem may be in something I didn't do or am looking at it incorrectly.

**Comment by UX3D-nopper** [comment]
Okay, I analyzed the model regarding geometry:
![grafik](https://user-images.githubusercontent.com/27856744/28476777-8baff224-6e51-11e7-989f-1ca674597901.png)
The screenshot is from the edit mode from the front roof. Also, I enabled face normal rendering. As you can see - the normals are not correct. The reason, why Blender renders this correct is, that the mesh is marked double sided.
I assume, the rest of the geometry artifacts are the same.
As described in the user documentation, the double sided flag comes from the material and not the mesh. 
https://github.com/KhronosGroup/glTF-Blender-Exporter/blob/master/docs/user.md
Thats the reason, why you have these artifacts.
How to solve this? Recalculate/set the normals at the rrenous place.

Appearance:
Also mentioned in the docs, the Blender material is not a PBR glTF 2,0 material and so no materials are exported. This means, the default material is used. So all color, texture information, alpha etc. gets lost.

To summarize:
- It is not a viewer and/or exporter bug.
- The scene needs some rework to render properly.

![grafik](https://user-images.githubusercontent.com/27856744/28477286-96fc0f3a-6e53-11e7-837a-321908a9d605.png)

**Comment by UX3D-nopper** [comment]
Thanks a lot that you took so much time to investigate this. Sorry, that I cannot help u further.

There will be an extension to glTF 2.0, called KHR_materials_cmnBlinnPhong. With this extension, it will also be possbile to export these materials as well.

Meanwhile, you need to convert the materials to PBR ones.

**Comment by DrX3D** [comment]
Thank you for your detailed response. That information helps a lot and I will be able to bring it to the Blender BOFs during SIGGRAPH. I see that knowing the target and how you are going to get there is still very important.

**Comment by UX3D-nopper** [comment]
Maybe you have a look at this:
https://github.com/KhronosGroup/glTF-Blender-Exporter/pull/53
We received a pull request, that as much PBR materials as possible are exported - without breaking the metallic roughness workflow.

**Comment by donmccurdy** [comment]
@MiguelAlgaba thanks for catching this — I've filed a new issue: https://github.com/KhronosGroup/glTF-Blender-Exporter/issues/199
